extension/view/js/InPageTranslation.js

```diff
@@ -567,17 +591,35 @@ class InPageTranslation {
 
             merge(node, scratch);
 
+
+            /*
+             * remove node again from nodesSent because someone might change
+             * the innerHTML or add children, and then we want to translate
+             * those.
+             * TODO: what if a node was mutated while translation was pending?
+             * Will that mutation then be ignored?
+             */
+            this.nodesSent.delete(node);
+
             /*
              * is this a good idea?
              * this.nodesSent.delete(node);
              * console.groupEnd(computePath(node));
              */
         };
 
-        this.updateMap.forEach(updateElement);
+        // pause observing mutations
+        this.stopMutationObserver();
+
+        try {
+            this.updateMap.forEach(updateElement);
+            this.updateMap.clear();
```

Comment:
@andrenatal quality estimation metrics got broken after cleaning of `updateMap` was moved up

Reply:
submit a PR